### ISSUE TITLE
Report Generation by School District

### ISSUE THREAD

**Issue by johntyree** [open-austin/texasappleseed#2]
#### Generate (printable?) report
- Discipline data
  - Data from ORRS (Open Record Requests)
    - What is this actually?
    - Let's define fields here
  - Maybe one of:
    - "We asked, but no response"
    - "Here's how you can ask"
    - Here's what you can can do...
## Open questions
- Who is the target audience for this?
- How do we intend this to be used?

**Comment by johntyree** [comment]
@MCraven2800

Do we have an example of what this might look like? Specifically it would be good to have an Idea for
- the format of the document
- the data that we expect to be in the document (Where does it currently live?)

**Comment by MCraven2800** [comment]
@johntyree 

let me check with Lauren---she mentioned designing a printable report a while ago.  If she doesn't have something I'll figure it out.

## Data in the document:
1. Total school enrollment
2. School enrollment by race/ethnicity
3. School enrollment by disability/special education status
4. School enrollment by "economically disadvantaged" status
5. School enrollment by identification as LGBTQ  (we will not get this data, it is just a wish)
6. Totals and breakdowns for the groups in 2-5 for:
   1. in-school suspensions
   2. out-of-school suspensions
   3. placement in Disciplinary Alternative Education Programs (DAEP)
   4. expulsions
7. Totals and breakdowns for the groups in 2-5 for:
   1. in-school arrests
   2. court referrals (like ticketing)
   3. Use of tasers and/or pepper spray
8. Information about the school police
   1. Whether a district has its own department or contracts with the local PD (and a link to the  
      contract between the ISD and PD, if available)
   2. Number of officers and any additional info about them
   3. Whether officers can use tasers and/or pepper spray
9.  Any unusual incidents that have happened recently (ie kid got choked, a bunch of students pepper sprayed, etc.)
10.  Three simple next steps for using the report (access other information, write a letter using a template on this site, speak at the next school board meeting and make some demands . . .)

It would also be great if we could place the color-coded ranking at the top of the report (based on the color of the district on the maps we create) or something that can designate if the district is particularly bad for students.

Items 1-6 can be found on the Texas Education Agency website --- http://tea.texas.gov/ --- under the Reports/Data tab.  We can send an Open Records Request to TEA if any of the data doesn't pop up in a useful format.

Items 1-8 are covered in what we've asked for in our Open Records Requests.  We will be getting that data in the next few weeks.

Item 9 is currently being compiled in a spreadsheet as incidents pop up in the media.

### Unrelated idea

We should have an Implicit Bias tab in the site!  Is that doable given all the other stuff we need to cover?  The section would cover implicit bias, how it works, debiasing techniques, etc.  Super important to understanding the pipeline.  If it can't have its own section, we can just incorporate it into the other sections.

---

jtyree (8-26): Edited for readability

**Comment by johntyree** [comment]
I just took a looked at the TEA site for a few minutes. Just long enough to get a CSV of discipline data and verify that it was usable. Am I correct in saying that it's not possible to just say "gimme all the data" ? It looks like we'll have to write something to automatically fill out the little forms to download the files for the various data sets (I was looking at the disciplinary data). The disciplinary data also appeared to be broken down by district at best. If we need anything more granular than that we'll have to talk about how to either back it out of the data by cross-referencing other sources or we'll need to ask TEA for it directly.

How long do we intend this report to be? One page? Will all of this fit on that page in a manageable way? (Possibly, yes? I don't know.)

Point 10 doesn't seem specific to any particular report. Is this just an extra sheet of information we'd like to be part of every report or does it just appear on the site somewhere?

The Implicit Bias page would not really be any additional work (for me, there's plenty of work there for you :smile:) and should be doable. Most of the site-building effort is in getting the layout right and making sure the embedded media stuff is functional, which we'll have to do anyway. It sounds like the Implicit Bias tab will mostly be written content that describes the idea, maybe with some visualizations but only to the extent that they help explain what it is and without needing them to be based on any live data. If so, then I would start putting that text together so we can include it in the prototype Patricia started at the end of our meeting. Feel free to create a new issue for that. Big green button at the top-right of this page.

**Comment by MCraven2800** [comment]
@johntyree 
Hello,

OK, I just went through this with Yamanda, but tell me if it seems like we need something else: We are going to ask TEA for a file with discipline data, broken down by campus (and within that by race, special education status and economic status) & with district as a field.

Yes, one page would be ideal, two at the most.   Point 10 would be part of every report.

**Comment by johntyree** [comment]
I don't know how explicit you can be about the data that we would get from them, but if you can ask specifically for CSV with columns as you stated them there please do. Getting a pile of oddly formatted excel files or a Microsoft Access database would make it much more difficult to get the data into a shape we can use.

Number of incidents per category will only be useful if we can compare them to the school population as a whole, right? So if they say, "10 incidents involving Latino students," we also need to know how many Latino students there were in total, or at least the fraction of the total student body that is Latino. This goes for the other classifications as well.

Gender isn't mentioned anywhere. Are we interested in that? I mean, I am :smiley_cat:, but is the organization?

**Comment by MCraven2800** [comment]
Your comparing question is an excellent one. I didn't think about it when I made the request bc I have been viewing student population data separately from discipline data on the tea website and just doing calculations when I need the info. Clearly not the easiest way!

I'll shoot the person who is getting the data for us an email to make sure they are including student population totals for the groups we are focusing on. 

Yes, definitely interested in gender. Here is exactly what we asked for:
1. School discipline data (in-school suspensions, out-of-school suspensions, DAEP placements, JJAEP placements, expulsions) for the past three years by grade level.  Data should be disaggregated by race/ethnicity, gender, special education status, limited English proficiency designation, and "economically disadvantaged" status.
2. School discipline data (in-school suspensions, out-of-school suspensions, DAEP placements, JJAEP placements, expulsions) for the past three years by school campus, with the district as a field (so that the campus data can be sorted by district).  Data should be disaggregated by race/ethnicity, gender, special education status, limited English proficiency designation, and "economically disadvantaged" status.

**Comment by MCraven2800** [comment]
Also, Yamanda may know if there's an easier way to do this if tea gets weird.

**Comment by johntyree** [comment]
That looks pretty good without knowing much more about it.

As far as the other point I made, I would think that asking for the student population as a whole, disaggregated in the same way, should take care of everything. Those are really the groups we're trying to compare when talking about implicit bias, I think, such that we can answer the question, "How does the group of students being disciplined differ from what one would expect if all of the students were being treated equally?"

**Comment by MCraven2800** [comment]
Agreed. I asked for the addition, I'll update on Monday. Thanks for catching that---very important!

**Comment by mscarey** [comment]
Here's a gist that shows [my attempt at putting TEA demographic and discipline data in one file](https://gist.github.com/mscarey/1b78deb0a6261b996f26). The end result was the file called [TX.json](https://gist.githubusercontent.com/mscarey/1b78deb0a6261b996f26/raw/42403dd592fa343f3ad6b2ef8b828f2047ded78c/TX.json). I tried to document the process so it could be replicated and modified. I was trying to make something suitable for a d3.js visualization, but I don't know whether that would ideally be a map or a chart.

**Comment by johntyree** [comment]
Nice. I was considering stuffing everything into a database but flat JSON will be easier to work with.

As far as generating actual downloadable reports, I was going to use the reportlab python library, which gives you a pdf canvas, more or less. They can be generated for each school district ahead of time and we can just serve them as static files. 

It would be nice to have some kind of plot on the actual page. For example, some bar charts of things we can know for certain, like how many of each type of disciplinary action occurred by gender. It's a bit harder to work with the other labels because they aren't mutually exclusive, where as we know every event will occur to either a male or female, so we can count those to get an accurate total. D3 should be suitable for this if you wanted to try making one.

**Comment by mscarey** [comment]
I ran into too many problems trying to manipulate the JSON, so I switched back to CSV to create the file with the ratios of disciplinary actions for different groups. I'm pretty sure we can feed CSV into D3 at least as easily as JSON. The CSV's in a pull request now. There's also a [new gist](https://gist.github.com/mscarey/bdec4603313dd81da530) documenting how I arrived at the CSV. There easily could be errors, so we should spot-check the CSV. If needed, I can also share the Jupyter workbook I was working in.

FWIW, I think the TEA treats the seven race categories as mutually exclusive too. In fact, I depended on that assumption in making the calculations in the CSV.
